packages/good-design/src/apps/goodid/idcard/GoodIdCard.tsx

```diff
@@ -75,7 +75,7 @@ const GoodIdCard = withTheme({ name: "GoodIdCard", skipProps: "credentialsList"
           {Object.entries(CredentialType).map(([key, credentialType]) => (
```

Comment:
you should use aggregated credentials hook which will provide you a single (latest issued) certificate per each credential type. please update

Reply:
The fetching of credentials (so the hook you mention) is not part of this component, that will be implemented by any parent displaying the card.
Here we map over all possible types (as of today, 4) because we show unverified or placeholder values when the certificate does not exist in the passed down credentiallist for the current connected user

Reply:
useAggredatedCredentials maps over all possible types and gives you a collection with one certificate per type. so please use it to fetch. thanks

Reply:
Again, not part of this component, will be implemented by a parent which uses good-id card
(considering different db solutions between gooddapp/wallet)

This specific map is purely a UI representation and will use the certificates passed down

Reply:
Because if there are no certificates to catch, the hook returns an empty array and I would still need this additional mapping to get to this view:

![image](https://github.com/GoodDollar/GoodWeb3-Mono/assets/6606028/304f24d2-c971-46c0-ac1d-db645f3415c7)

Reply:
@L03TJ3 I've adjusted hook to return all 4 types. each item will contain unique key to list rendering, type and typeName (to display as label) and **optional** `certificate` property. if it exists and non-undefined, that means user have certificate of the corresponding type, if not  you show `-` as in the design below

Reply:
@L03TJ3 if you need it as type => item object map you could use lodash's `keyBy` at Your component

Reply:
@L03TJ3 when will you update it ?